sonar-cloud-pipeline.yml

```diff
@@ -18,13 +19,32 @@ jobs:
       wget -q https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_VERSION}-linux.zip
       unzip -q sonar-scanner-cli-${SONAR_VERSION}-linux.zip
 
+      pip install --user -r requirements.txt
```

Comment:
I'm tempted to say move this into a separate script file since its looking like a lot of bash written in a config file. Up to you.

Reply:
Yeah, I was feeling that way too. I've gone ahead and split the coverage stuff out -- it would be nice to be able to run that same script locally while I'm working on improving coverage, which won't work currently but I'll probably do that in a followup change.